Ice-shelf solo driver and MISMIP+ updates

alex-huth opened the pull request:
-Several edits to the ice shelf solo driver so that it works with the rest of the current MOM6
-Added capability to initialize a surface mass balance (SMB) that is held contstant over time when running from the ice-shelf solo driver (see new subroutine initialize_ice_SMB). This is required for MISMIP+. A constant SMB can also be used from the MOM driver for coupled ice-shelf/ocean experiments (e.g. MISOMIP).
-The new, constant SMB is passed into solo_step_ice_shelf, where change_thickness_using_precip is called
-Added capability to save both non-time-stamped and time-stamped restart files when using the ice shelf solo driver. This is useful for debugging.
-slight reorganization to when ice shelf post_data calls are made

Hallberg-NOAA reviewed (commented):
I do not understand why we need a separate `solo_ice_shelf_diag_mediator_end()` routine in MOM_diag_mediator.F90 that is distict from `diag_mediator_end()`.  They appear to me to take the same arguments and do the same thing, perhaps with a little error checking.  Given that they are so similar, why not add the safety checks to `diag_mediator_end()`?

Also, the capability being added with the new runtime parameter SAVE_BOTH_RESTARTS could already have been obtained by setting RESTART_CONTROL=3.  The new parameter may be more explicit, but it does add to the list of parameters without actually adding any new capabilities.

alex-huth commented:
I revised the PR as suggested by @Hallberg-NOAA . I added safety checks to `diag_mediator_end()` so that the separate  `solo_ice_shelf_diag_mediator_end()` routine is no longer necessary. I also removed the SAVE_BOTH_RESTARTS option, which was useful during development, but is not needed anymore.

Hallberg-NOAA reviewed (approved):
This set of code changes looks reasonable to me.
